src/libtls/tls_server.c

```diff
@@ -456,43 +478,66 @@ static status_t process_client_hello(private_tls_server_t *this,
 		enumerator = this->crypto->create_ec_enumerator(this->crypto);
 		while (enumerator->enumerate(enumerator, &group, &curve))
 		{
+			if (!this->requested_curve &&
+				curve != key_type &&
+				peer_supports_curve(this, curve))
+			{
+				this->requested_curve = curve;
+			}
 			if (curve == key_type && peer_supports_curve(this, curve))
 			{
+				DBG1(DBG_TLS, "key share agreed on %N",
```

Comment:
This log message sounds a bit weird.

Reply:
```suggestion
				DBG1(DBG_TLS, "DH group %N negotiated",
```
what do you think about this?

Reply:
`using key exchange %N`